tests/unit/test_routes/test_resource_retreival.py

```diff
@@ -0,0 +1,186 @@
+
+from configs import PaginatorConfig
+from datetime import datetime, timedelta
+from app.utils import get_error_code_from_status
+from app.versioning import LATEST_API_VERSION
+from yaml import load
+
+
+# TODO: We need negative unit tests (what happens when bad data is sent)
```

Comment:
We should get rid of this TODO and make sure that we have negative unit tests before we merge